cinema/serializers.py

```diff
@@ -132,6 +149,31 @@
                 Ticket.objects.create(order=order, **ticket_data)
             return order
 
+    def update(self, instance, validated_data):
+        tickets_data = validated_data.pop("tickets")
+        instance = super().update(instance, validated_data)
+
+        keep_tickets = []
+        for ticket_data in tickets_data:
+            if "id" in ticket_data.keys():
+                if Ticket.objects.filter(id=ticket_data["id"]).exists():
+                    ticket = Ticket.objects.get(id=ticket_data["id"])
+                    ticket.row = ticket_data.get("row", ticket.row)
+                    ticket.seat = ticket_data.get("seat", ticket.seat)
+                    ticket.save()
+                    keep_tickets.append(ticket.id)
+                else:
+                    continue
+            else:
+                ticket = Ticket.objects.create(order=instance, **ticket_data)
+                keep_tickets.append(ticket.id)
+
+        for ticket in instance.tickets.all():
+            if ticket.id not in keep_tickets:
+                ticket.delete()
```

Comment:
The loop for deleting tickets that are not in `keep_tickets` is correct, but ensure that this logic aligns with the business requirements for handling orphaned tickets.